### DISCUSSION TITLE
Make Every Merchant a Bitcoin Vendor - 4M Sats

### DISCUSSION THREAD

**Discussion by pretyflaco** [GaloyMoney/galoy#1418] [Bounties]
## Description
This bounty is complementary to [LNURL-withdraw (LN Wallet)](https://github.com/GaloyMoney/galoy/discussions/1417). Enable Galoy wallet users to generate LNURL-withdraw QR or string that `LN Wallet` can scan to receive funds easily.
In the [LNURL specs](https://github.com/fiatjaf/lnurl-rfc/blob/luds/03.md) this function is referred to as `LN Service`.

## Background

`LN Service` allows a user to generate a QR or string that can be passed on to a payee `LN Wallet` with LNURL-withdraw support. 

This would allow anyone with the blink app to become a bitcoin vendor. The generated QR could be printed with a [thermal printer](https://www.amazon.com/dp/B09C25H9T3/ref=sspa_dk_hqp_detail_aax_0?psc=1&sp_csd=d2lkZ2V0TmFtZT1zcF9ocXBfc2hhcmVk&spLa=ZW5jcnlwdGVkUXVhbGlmaWVyPUExWUhZRlRKVlVZTkpFJmVuY3J5cHRlZElkPUEwNjgyODYxWUxSVlFFWExOWDNSJmVuY3J5cHRlZEFkSWQ9QTA3MTkwOTZOOExRQ1NWWlMyTTkmd2lkZ2V0TmFtZT1zcF9ocXBfc2hhcmVkJmFjdGlvbj1jbGlja1JlZGlyZWN0JmRvTm90TG9nQ2xpY2s9dHJ1ZQ==) directly from the phone and sold for cash over the counter.

If merchants could easily sell bitcoin over the counter for cash, it would

- add new revenue stream (small commission for sale)
- bring new customers who want to buy bitcoin (or US Dollars*)
- incentivize merchant to accept bitcoin for goods & services (replenishing bitcoin stock)
- offer kyc-free bitcoin / US Dollars* to customers
- make bitcoin more easily accessible in more areas
- Improve a circular economy around BTC as a local currency.

*If customer uses a LN Wallet with Stablesats, they can choose to hold the purchased bitcoin as US Dollars in their wallet

Furthermore, being able to generate an LNURL-withdraw QR & string would allow paying with the link while offline by loading it to an NFC sticker or card, as well as for contactless support or just support offline usage of the wallet app as described [here](https://github.com/GaloyMoney/galoy/discussions/1351) 

Useful docs:
[LUD-03](https://github.com/fiatjaf/lnurl-rfc/blob/luds/03.md)
[LUD-08](https://github.com/fiatjaf/lnurl-rfc/blob/luds/08.md)
[LUD-17](https://github.com/fiatjaf/lnurl-rfc/blob/luds/17.md)
[LNPay Docs](https://docs.lnpay.co/api/lnurl-withdraw/how-does-lnurl-withdraw-work)
[BOLT.FUN Docs](https://bolt.fun/guide/web-services/lnurl/withdraw)
[LNBits](https://github.com/lnurl/luds/blob/legacy/lnurl-withdraw.md)
## Task

Submit a PR that enables above outlined functionality.

## Definition of Done

PR gets merged and enables Galoy wallet users to generate LNURL-withdraw QR / string.

## Bounty
4M Sats (0.04 BTC)

## Bounty Patrons
[Galoy](https://galoy.io/)

**Comment by kngako** [comment]
Anyone working on this?
